tycho-gpg-plugin/src/main/java/org/eclipse/tycho/gpg/SignRepositoryArtifactsMojo.java

```diff
@@ -0,0 +1,153 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat Inc. and others.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.tycho.gpg;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Arrays;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.gpg.AbstractGPGMojoExtension;
+import org.apache.maven.plugin.gpg.ProxySignerWithPublicKeyAccess;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.archiver.Archiver;
+import org.codehaus.plexus.archiver.ArchiverException;
+import org.codehaus.plexus.archiver.UnArchiver;
+import org.codehaus.plexus.archiver.xz.XZArchiver;
+import org.codehaus.plexus.archiver.xz.XZUnArchiver;
+import org.codehaus.plexus.archiver.zip.ZipArchiver;
+import org.codehaus.plexus.archiver.zip.ZipUnArchiver;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.codehaus.plexus.util.xml.Xpp3DomBuilder;
+import org.codehaus.plexus.util.xml.Xpp3DomWriter;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.eclipse.equinox.p2.repository.artifact.IArtifactDescriptor;
+
+/**
+ * Modifies the p2 metadata (<code>artifacts.xml</code>) to add PGP signatures for each included
+ * artifact. Signatures are added as <code>pgp.signatures</code> property, in armored form.
+ */
+@Mojo(name = "sign-p2-artifacts", requiresProject = true, defaultPhase = LifecyclePhase.PREPARE_PACKAGE)
+public class SignRepositoryArtifactsMojo extends AbstractGPGMojoExtension {
```

Comment:
Instead of SignRepositoryArtifactsMojo > AbstractGPGMojoExtension>AbstractGpgMojo maybe AbstractGPGMojoExtension could be left out here or is it likeley that there are more signers required?

If yes instead of AbstractGPGMojoExtension maybe "AbstractPublicKeyGPGMojo" would be a better name

Reply:
I'll look at that.

Reply:
So I looked at it, and the issue is that package: in order to reuse some pieces of the AbstractGpgMojo which are package protected, we have to use the same package name. I prefer keeping the intermediary class to "open" some package-protected content and keeping the actual business code of Tycho in a mojo of the tycho package.

Reply:
Maybe it would be good to open a patch at maven then to allow more open reuse there also?

Reply:
yes, it would be good. However, it's not of high enough importance to me at the moment to deal wih it right now.